### PR TITLE
[#5333 6/6] Region snapshot replacement finish

### LINKED ISSUE #5333: implement region replacement: read-only volumes
This issues covers the case where we need to replace a ROP in a virtual disk

Comments from 3886 are copied here:

If you have a ROP that needs a downstairs replacement, Nexus needs to follow these rough steps:

* Create a new blank region with the same dimensions (block size, extent size, extent_count) as the one you wish to replace.
* Pick from the ROP, one of the working downstairs targets that will be used as the source data.
* Restart the newly created region with the "clone" subcommand, giving it the repair address of the working (but RO) downstairs target.
* Let the clone finish, the replacement downstairs will exit.
* Restart the replacement downstairs in RO.
* Replace the ROP target that was bad with the IP:Port of the new replacement.

I realize there are a lot of details to work out here, but that's the general idea.

### PR DESCRIPTION
**Pull request by jmpesp**
For the final commit in this series, add a background task that will detect when all the related steps of a region snapshot replacement are done, and then mark that region snapshot replacement request as Complete.

Closes #5333.

### CODE REVIEW

File: nexus/src/app/background/tasks/region_snapshot_replacement_finish.rs [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                .in_progress_region_snapshot_replacement_steps(
+                    opctx, request.id,
+                )
+                .await
+            {
+                Ok(count) => count,
+
+                Err(e) => {
+                    let s = format!(
+                        "counting non-complete region snapshot replacement \
```
Comment (andrewjstone) [inline nexus/src/app/background/tasks/region_snapshot_replacement_finish.rs:70]: nit: "incomplete" ?

**Review by andrewjstone — APPROVED** [review]
Ship it! Congrats on the huge accomplishment! 🎉

File: nexus/src/app/background/tasks/region_snapshot_replacement_step.rs [hunk 2]
```diff
@@ -413,12 +413,6 @@ impl BackgroundTask for RegionSnapshotReplacementFindAffected {
         opctx: &'a OpContext,
     ) -> BoxFuture<'a, serde_json::Value> {
         async move {
-            let log = &opctx.log;
-            info!(
```
Comment (leftwo) [inline nexus/src/app/background/tasks/region_snapshot_replacement_step.rs:417]: These were the logs that were too noisy?

**Review by leftwo — APPROVED** [review]

File: nexus/src/app/background/tasks/region_snapshot_replacement_step.rs (hunk 2 above)
Comment (jmpesp) [reply nexus/src/app/background/tasks/region_snapshot_replacement_step.rs:417]: Yeah, see https://github.com/oxidecomputer/omicron/pull/6323

File: nexus/src/app/background/tasks/region_snapshot_replacement_finish.rs (hunk 1 above)
Comment (jmpesp) [reply nexus/src/app/background/tasks/region_snapshot_replacement_finish.rs]: done in 86b590a07, thanks :)

**Pull request merged**